hedera-node/hedera-token-service-impl/src/main/java/com/hedera/node/app/service/token/impl/handlers/TokenUpdateHandler.java

```diff
@@ -37,15 +53,40 @@ public class TokenUpdateHandler implements TransactionHandler {
      * <p>Please note: the method signature is just a placeholder which is most likely going to
      * change.
      *
-     * @param txBody the {@link TransactionBody} with the transaction data
+     * @param txn the {@link TransactionBody} with the transaction data
      * @param payer the {@link AccountID} of the payer
      * @return the {@link TransactionMetadata} with all information that needs to be passed to
      *     {@link #handle(TransactionMetadata)}
      * @throws NullPointerException if one of the arguments is {@code null}
      */
     public TransactionMetadata preHandle(
-            @NonNull final TransactionBody txBody, @NonNull final AccountID payer) {
-        throw new UnsupportedOperationException("Not implemented");
+            @NonNull final TransactionBody txn,
+            @NonNull final AccountID payer,
+            @NonNull final AccountKeyLookup keyLookup,
+            @NonNull final ReadableTokenStore tokenStore) {
+        final var op = txn.getTokenUpdate();
+        final var tokenId = op.getToken();
+        final var meta =
+                new SigTransactionMetadataBuilder(keyLookup).payerKeyFor(payer).txnBody(txn);
+        final var tokenMeta = tokenStore.getTokenMeta(tokenId);
+        if (tokenMeta.failed()) {
+            meta.status(tokenMeta.failureReason());
+        } else {
+            final var tokenMetadata = tokenMeta.metadata();
+            final var adminKey = tokenMetadata.adminKey();
+            adminKey.ifPresent(meta::addToReqNonPayerKeys);
+            if (op.hasAutoRenewAccount()) {
+                meta.addNonPayerKey(op.getAutoRenewAccount(), INVALID_AUTORENEW_ACCOUNT);
+            }
+            if (op.hasTreasury()) {
+                meta.addNonPayerKey(op.getTreasury());
+            }
+            if (op.hasAdminKey()) {
+                final var newAdminKey = asHederaKey(op.getAdminKey());
+                newAdminKey.ifPresent(meta::addToReqNonPayerKeys);
+            }
+        }
```

Comment:
Personally I prefer a `return` statement as opposed to a big if-else, but not a huge deal
e.g.
```
if (tokenMeta.failed()) {
    return meta.status(tokenMeta.failureReason()).build();
}
// no 'else' needed

Reply:
Agreed, changed!